api/envoy/extensions/filters/http/header_mutation/v3/header_mutation.proto

```diff
@@ -15,12 +16,49 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#protodoc-title: Header mutation filter configuration]
 // [#extension: envoy.filters.http.header_mutation]
 
+message QueryParameterValueOption {
+  // Describes the supported actions types for query parameter append action.
+  enum QueryParameterAppendAction {
+    // If the query parameter already exists, this action will result in:
+    //
+    // - Duplicate query parameter keys in the HTTP path.
+    //
+    // If the query parameter doesn't exist then this will add new query parameter with specified key and value.
+    APPEND_IF_EXISTS_OR_ADD = 0;
+
+    // This action will add the query parameter if it doesn't already exist. If the query parameter already exists
+    // then this will be a no-op.
+    ADD_IF_ABSENT = 1;
+
+    // This action will overwrite the specified value by discarding any existing values if the query parameter already
+    // exists. If the query parameter doesn't exist then this will add the query parameter with specified key and
+    // value.
+    OVERWRITE_IF_EXISTS_OR_ADD = 2;
+
+    // This action will overwrite the specified value by discarding any existing values if the query parameter already
+    // exists. If the query parameter doesn't exist then this will be no-op.
+    OVERWRITE_IF_EXISTS = 3;
+  }
+
+  QueryParameterAppendAction append_action = 1;
+
+  // Query parameter that this operation applies to.
+  config.core.v3.QueryParameter query_parameter = 2;
```

Comment:
nit: one thing about the naming, I think the `query` is enough. The `query` should no several possible meanings in this context.

Reply:
do you just mean this variable or are you also referring to `QueryParameterValueOption` -> `QueryValueOption`, `QueryParameterMutation` -> `QueryMutation` etc? I was opting to be extra-explicit on the config api and looser in the code, but I get what you mean.